models/repo_permission.go

```diff
@@ -16,6 +16,12 @@ func (p *Permission) IsOwner() bool {
 	return p.AccessMode >= AccessModeOwner
 }
 
+// IsWriter checks if the current user has write or higher access to the
+// repository.
+func (p *Permission) IsWriter() bool {
+	return p.AccessMode >= AccessModeWrite
```

Comment:
This method have been deleted since we should use `CanWrite(models.UnitTypeIssues)` instead.

Reply:
@lunny Curiosity as I am not familiar with that part of the code. But do I need to use since both issues and PRs can be locked

```
CanWrite(models.UnitTypeIssues) && CanWrite(models.UnitTypePullRequests)
```

Reply:
@adelowo You should add a `if`:
```go
if issue.IsPull {
return CanWrite(models. UnitTypePullRequests)
}
return CanWrite(models. UnitTypeIssues)
```